File: spring-email-java/src/main/java/com/github/migangqui/spring/email/bean/SendEmailResult.java
package com.github.migangqui.spring.email.bean;

import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
@JsonInclude(NON_NULL)
public class SendEmailResult {
	private int status;
	private String cause;
	private Exception exception;
}

File: spring-email-kotlin/src/main/kotlin/com/github/migangqui/spring/email/bean/SendEmailResult.kt
package com.github.migangqui.spring.email.bean

import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL
import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(NON_NULL)
data class SendEmailResult(
    val status: Int = 0,
    val cause: String? = null,
    val exception: Exception? = null
)
